build-pipeline/linux-pipeline.groovy

```diff
@@ -0,0 +1,13 @@
+@Library('dotnet-ci') _
+
+simpleNode ('Ubuntu16.04', 'latest-or-auto-docker') {
+    stage ('Checking out source') {
+        checkout scm
+    }
+    stage ('Install pwsh') {
```

Comment:
This is modifying the state of the machine.  Should this get cleaned up at the end?  An alternative would be to utilize a Docker container.

Reply:
Can it be installed to the local workspace directory?  That would also solve this issue.

Reply:
Good point. I didn't know if the agents were going to sandbox test runs like Travis/AppVeyor do. I think it should be possible to download as a tarball as long as these dependencies are installed:

libc6, libgcc1, libgssapi-krb5-2, liblttng-ust0, libstdc++6, libcurl3, libunwind8, libuuid1, zlib1g, libssl1.0.0, libicu55

https://github.com/PowerShell/PowerShell/blob/master/docs/installation/linux.md#binary-archives

Reply:
I think those should be available, if not we can get them.